### ISSUE TITLE
Registering Facade inside AddOn

### ISSUE THREAD

**Issue by gochev** [Adyen/adyen-hybris#210]
**Describe the bug**
Found in version 6.4 In adyenv6b2ccheckoutaddon-spring.xml you are registering / overriding a facade.

	<alias name="adyenCheckoutFlowFacade" alias="checkoutFlowFacade"/>
	<bean id="adyenCheckoutFlowFacade" class="com.adyen.v6.acceleratorfacades.flow.impl.AdyenCheckoutFlowFacade" parent="defaultCheckoutFlowFacade" />

Since addons are installed on TOP of the storefront accelerator this overrides any client checkoutFlowFacades with yours basically breaking any client flow facade in existence. You should never REGISTER a facades as part of an addon, this is extremely bad from a quality perspective and extremely hard to find. 

**Expected behavior**
To not override client extension facades or storefront facades

**Additional context**

In Hybris there are 2 facades CheckoutFacade and CheckoutFlowFacade both are overwritten in 99% of the projects, by introducing this in an addon you basically override one of them in an extremely hidden way basically replacing delivery modes and delivery address. It is even worse because in your addon you are overriding getCheckoutCart and calling getCartFacade().getSessionCart() however even this cartFacade is not the client one, you not only break the flow facade but also not use the client override facade to take the session cart instead you are using the default one.

**Comment by gochev** [comment]
Also the comment on top 



Is very invalid.... because the DefaultAcceleratorCheckoutFacade have this method with exact same body. I am not sure is this even reviewed ? :)

**Comment by rkewlani** [comment]
Hi @gochev,
Thanks for writing to us. We are looking into it.

Regards,
Adyen

**Comment by martinsrenato** [comment]
Hi @gochev ,

The main reason why we add this facade is to override the `getCheckoutCart` method so it does not retrieve _paymentInfo_ using the default Hybris classes and implementation. The `DefaultCheckoutFacade` implementation for this method is the one below, and assumes that _paymentInfo_ is an instance of _CreditCardPaymentInfoModel_, a model that would be compatible with payment methods supported by Adyen.
```java
@Override
public CartData getCheckoutCart()
{
final CartData cartData = getCartFacade().getSessionCart();
if (cartData != null)
{
	cartData.setDeliveryAddress(getDeliveryAddress());
	cartData.setDeliveryMode(getDeliveryMode());
	cartData.setPaymentInfo(getPaymentDetails());
}
return cartData;
}

(...)

protected CCPaymentInfoData getPaymentDetails()
{
final CartModel cart = getCart();
if (cart != null)
{
	final PaymentInfoModel paymentInfo = cart.getPaymentInfo();
	if (paymentInfo instanceof CreditCardPaymentInfoModel)
	{
		return getCreditCardPaymentInfoConverter().convert((CreditCardPaymentInfoModel) paymentInfo);
	}
}
return null;
}
```
Since the plugin's goal is to provide an easy way to integrate with Adyen's services, we need full control of what is set on paymentInfo and for that we need to override some of the logic from the default hybris classes. 
We investigated on how to do that without changing the checkoutFlowFacade, but no successful alternative was found.

One small change that we did based on your comments was removing the `hasNoPaymentInfo` method from `AdyenCheckoutFlowFacade` (check PR #261). It had the exact same body as the default hybris implementations, as you mentioned, so there was no need to override it.

As always, feel free to suggest any other ideas or code that you think could work better on this case.

Regards,
Adyen
